spec/example_app/spec/models/product_spec.rb

```diff
@@ -0,0 +1,20 @@
+require "rails_helper"
+
+RSpec.describe Product do
+  describe "validations" do
+    it { should allow_value("http://example.com/foo/bar").for(:image_url) }
+    it { should allow_value("https://example.com/foo/bar").for(:image_url) }
+    it { should_not allow_value("ftp://example.com/foo/bar").for(:image_url) }
+  end
+
+  describe "#image_url" do
+    it "is trimmed on save" do
+      attrs = FactoryBot.attributes_for(
+        :product,
+        image_url: "\n https://example.com/foo/bar  \n",
+      )
+      product = Product.create(attrs)
```

Comment:
```suggestion
    it "is trimmed on save" do
      product = create(
      :product,
      image_url: "\n https://example.com/foo/bar  \n",
    )
```

I don't seem to be able to get the suggestion to work out, but I hope you get the idea 😅 

What do you think about collapsing all of these together?

Reply:
Yeah, I don't know why I went with that. It's something about FactoryBot not doing anything clever behind my back, but it's a bit silly. OK, I'll fix it :+1: